driver/sccb.c

```diff
@@ -38,47 +38,64 @@ static const char* TAG = "sccb";
 #define ACK_VAL                 0x0                   /*!< I2C ack value */
 #define NACK_VAL                0x1                   /*!< I2C nack value */
 #if CONFIG_SCCB_HARDWARE_I2C_PORT1
-const int SCCB_I2C_PORT         = 1;
+const int SCCB_I2C_PORT_DEFAULT = 1;
 #else
-const int SCCB_I2C_PORT         = 0;
+const int SCCB_I2C_PORT_DEFAULT = 0;
 #endif
 
+static int sccb_i2c_port;
+static bool sccb_owns_i2c_port;
+
 int SCCB_Init(int pin_sda, int pin_scl)
 {
     ESP_LOGI(TAG, "pin_sda %d pin_scl %d", pin_sda, pin_scl);
     i2c_config_t conf;
     memset(&conf, 0, sizeof(i2c_config_t));
+
+    sccb_i2c_port = SCCB_I2C_PORT_DEFAULT;
+    sccb_owns_i2c_port = true;
+    ESP_LOGI(TAG, "sccb_i2c_port=%d\n", sccb_i2c_port);
+
     conf.mode = I2C_MODE_MASTER;
     conf.sda_io_num = pin_sda;
     conf.sda_pullup_en = GPIO_PULLUP_ENABLE;
     conf.scl_io_num = pin_scl;
     conf.scl_pullup_en = GPIO_PULLUP_ENABLE;
     conf.master.clk_speed = SCCB_FREQ;
 
-    i2c_param_config(SCCB_I2C_PORT, &conf);
-    i2c_driver_install(SCCB_I2C_PORT, conf.mode, 0, 0, 0);
-    return 0;
+    if(i2c_param_config(sccb_i2c_port, &conf) != ESP_OK) {
+        return ESP_ERR_INVALID_ARG;
+    }
+    if(i2c_driver_install(sccb_i2c_port, conf.mode, 0, 0, 0) != ESP_OK) {
+        return ESP_FAIL;
```

Comment:
likewise, can we return the value returned by i2c_driver_install?

Reply:
Done.